src/useAsync/index.ts

```diff
@@ -120,24 +125,25 @@ function useAsync<Result = any>(
     // 确保不会返回被取消的结果
     const runCount = count.current;
     set(s => ({ ...s, loading: true }));
+    params.current = args;
     return fn(...args)
       .then(data => {
-        if (runCount === count.current) {
+        if (!autoCancel || runCount === count.current) {
           if (_options.onSuccess) {
-            _options.onSuccess(data);
+            _options.onSuccess(data, args || []);
           }
-          if (runCount === count.current) {
+          if (!autoCancel || runCount === count.current) {
```

Comment:
It is best to add a note here.
because it‘s consistent with the above conditions.
it is easy to think that it is redundant.